sentry-options-validation/src/lib.rs

```diff
@@ -303,6 +310,86 @@ impl Default for SchemaRegistry {
     }
 }
 
+/// Watches the values file for changes, reloading if there are any
+/// Uses polling for now, could use `inotify` or similar later on
+pub struct ValuesWatcher {
+    stop_signal: Arc<AtomicBool>,
+    thread: Option<JoinHandle<()>>,
+}
+
+impl ValuesWatcher {
+    /// Creates a new ValuesWatcher struct and spins up the watcher thread
+    ///
+    /// Returns a FileRead error if we fail to get metadata for the file
+    /// Thread spawning could panic but that would be really bad
+    pub fn new(path: &Path) -> ValidationResult<Self> {
+        // validate permissions and existence of file
+        fs::metadata(path)?;
+
+        let stop_signal = Arc::new(AtomicBool::new(false));
```

Comment:
Are we handling synchronization when multiple file watcher threads are brought up?